src/NetTopologySuite/Utilities/PriorityQueue.cs

```diff
@@ -87,6 +88,15 @@ public T Peek()
                 : node.Data;
         }
 
+        public bool RemoveIf(Predicate<T> predicate)
+        {
+            var items = _queue.Where(t => t != null ? predicate(t.Data) : false) ?? Array.Empty<PriorityQueueNode<T, T>>();
```

Comment:
If you don't want to either defer adding `RemoveIf` by pulling in locationtech/jts@88d5965 early or else work on that other version of `RemoveIf` that I had in the comments, then we should at least do something like:
```suggestion
            var items = _queue.Where(t => t != null ? predicate(t.Data) : false).ToArray() ?? Array.Empty<PriorityQueueNode<T, T>>();
```
It's possible that the one caller will never care, and I have no idea why this isn't throwing any exceptions as-written, but nodes are getting missed this way.  The following test fails unless I either add the `.ToArray()` call above, or rewrite it the way I had in my non-review comment on the PR:
```csharp
[Test]
public void TestOrder2()
{
    var q = new PriorityQueue<int>();
    q.Add(1);
    q.Add(10);
    q.Add(5);
    q.Add(8);
    q.Add(-1);
    q.RemoveIf(_ => true);
    Assert.That(q.IsEmpty);
}
```

Reply:
Having `RemoveIf` in place or not does not have any effect on the outcome.

> I have ported NTS' `AlternativePriorityQueue<TPriority, TData>` and related `PriorityQueueNode<TPriority, TData>` class to java and used it within JTS. Both implementations yield the same results now.

I had to use the following tweaks on the `Corner` class (both NTS and JTS) to __ensure__ stable ordering within the queue:
```patch
 src/NetTopologySuite/Simplify/Corner.cs | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/NetTopologySuite/Simplify/Corner.cs b/src/NetTopologySuite/Simplify/Corner.cs
index c1e5f6ad..cfc27283 100644
--- a/src/NetTopologySuite/Simplify/Corner.cs
+++ b/src/NetTopologySuite/Simplify/Corner.cs
@@ -6,11 +6,14 @@ namespace NetTopologySuite.Simplify
     internal sealed class Corner : IComparable<Corner>
     {
 
+        private static volatile long Counter = long.MinValue;
+
         private readonly LinkedLine _edge;
         private readonly int _index;
         private int _prev;
         private int _next;
         private double _area;
+        private readonly long _itemNo;
 
         public Corner(LinkedLine edge, int i)
         {
@@ -19,6 +22,8 @@ public Corner(LinkedLine edge, int i)
             _prev = edge.Prev(i);
             _next = edge.Next(i);
             _area = AreaFn(edge, i);
+
+            _itemNo = Counter++; ;
         }
 
         public bool IsVertex(int index)
@@ -55,7 +60,9 @@ private static double AreaFn(LinkedLine edge, int index)
          */
         public int CompareTo(Corner o)
         {
-            return _area.CompareTo(o._area);
+            int res = _area.CompareTo(o._area);
+            if (res != 0) return res;
+            return _itemNo.CompareTo(o._itemNo);
         }
 
         public Envelope Envelope

```